src/main/java/ru/ylab/web/controllers/AdminUsersController.java

```diff
@@ -0,0 +1,106 @@
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.InitBinder;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import ru.ylab.aspects.LogRequest;
+import ru.ylab.dto.in.UserForm;
+import ru.ylab.dto.in.UserSearchForm;
+import ru.ylab.dto.mappers.UserMapper;
+import ru.ylab.dto.out.UserDto;
+import ru.ylab.services.entities.UserService;
+import ru.ylab.services.validation.UserFormValidator;
+
+import static ru.ylab.utils.constants.WebConstants.ADMIN_URL;
+import static ru.ylab.utils.constants.WebConstants.ID_URL;
+import static ru.ylab.utils.constants.WebConstants.SEARCH_URL;
+import static ru.ylab.utils.constants.WebConstants.USERS_URL;
+
+/**
+ * Controller for handling users HTTP requests for admin.
+ *
+ * @author azatyamanaev
+ */
+@LogRequest
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(ADMIN_URL + USERS_URL)
+public class AdminUsersController {
+
+    /**
+     * Instance of an {@link UserMapper}.
+     */
+    private final UserMapper userMapper;
+
+    /**
+     * Instance of an {@link UserService}.
+     */
+    private final UserService userService;
+
+    /**
+     * Instance of an {@link UserFormValidator}.
+     */
+    private final UserFormValidator userFormValidator;
+
+    @InitBinder
+    public void initBinder(WebDataBinder binder) {
+        binder.addValidators(userFormValidator);
+    }
+
+    /**
+     * Gets user for admin and writes it to response.
+     */
+    @GetMapping(ID_URL)
+    public ResponseEntity<UserDto> getUser(@PathVariable("id") Long id) {
+        UserDto dto = userMapper.mapToDto(userService.get(id));
+        return ResponseEntity.ok(dto);
+    }
+
+    /**
+     * Gets users for admin and writes them to response.
+     */
+    @GetMapping
+    public ResponseEntity<List<UserDto>> getUsers() {
+        List<UserDto> dtos = userMapper.mapToDto(userService.getAll());
+        return ResponseEntity.ok(dtos);
+    }
+
+    /**
+     * Searches users for admin and writes them to response.
+     */
+    @GetMapping(SEARCH_URL)
+    public ResponseEntity<List<UserDto>> searchUsers(UserSearchForm form) {
+        List<UserDto> dtos = userMapper.mapToDto(userService.searchUsers(form));
+        return ResponseEntity.ok(dtos);
+    }
+
+    /**
+     * Creates user for admin and sets created response status.
+     */
+    @PostMapping
+    public ResponseEntity<String> createUser(@Validated @RequestBody UserForm form) {
+        userService.createByAdmin(form);
+        return new ResponseEntity<>("", HttpStatus.CREATED);
```

Comment:
Можно возвращать не строку, а Void с статусом CREATED.

Reply:
[return void instead of empty string for 201, 204 http statuses](https://github.com/azatyamanaev/y_lab_java/pull/4/commits/b8383d49e9e958d32a5854f47b1449a5f40d5722)